CRM/Civicase/Api/Wrapper/CaseList.php

```diff
@@ -1,149 +1,153 @@
+          'sort' => 'start_date',
+          'display_type' => 'date',
+        ),
+        array(
+          'name' => 'modified_date',
+          'label' => ts('Last Updated'),
+          'sort' => 'modified_date',
+          'display_type' => 'date',
+        ),
+        array(
+          'name' => 'myRole',
+          'label' => ts('My Role'),
+          'sort' => 'my_role.label_b_a',
+          'display_type' => 'multiple_values',
+        ),
+      ),
+    );
+  }
+
+  /**
+   * Returns list of cases to be shown on case lists.
+   *
+   * @param array $params
+   *   Parameters array for the API call.
+   *
+   * @return array
+   *   Result with the list of cases to be shown.
+   */
+  public function getCaseList(array $params) {
+    $loggedContactID = CRM_Core_Session::singleton()->getLoggedInContactID();
+
+    $defaultAPIReturnedColumns = array(
+      'subject', 'case_type_id', 'status_id', 'is_deleted', 'start_date',
+      'modified_date', 'contacts', 'activity_summary', 'category_count',
+      'tag_id.name', 'tag_id.color', 'tag_id.description',
+    );
+    $params['return'] = (isset($params['return']) ? array_merge($defaultAPIReturnedColumns, $params['return']) : $defaultAPIReturnedColumns);
+    $cases = civicrm_api3('Case', 'getdetails', $params);
+
+    foreach ($cases['values'] as &$case) {
+      $caseLockedContacts = civicrm_api3('CaseContactLock', 'get', array(
+        'case_id' => $case['id'],
+        'contact_id' => $loggedContactID,
+      ));
+
+      // If case is locked for current user, activities should not be sent in
+      // response.
+      if ($caseLockedContacts['count'] > 0) {
+        $case['activity_summary'] = array();
+        $case['lock'] = 1;
+      }
+      else {
+        $case['lock'] = 0;
+      }
+
+      foreach ($case['contacts'] as $contact) {
+        if ($contact['manager'] == 1) {
+          $case['manager'] = $contact;
+        }
+
+        if ($loggedContactID == $contact['contact_id']) {
+          $case['myRole'][] = $contact['role'];
+        }
+      }
+
+      $case['next_activity'] = CRM_Utils_Array::value(0, $case['activity_summary']['next']);
+    }
+
+    return $cases;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fromApiInput($apiRequest) {
+    return $apiRequest;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function toApiOutput($apiRequest, $result) {
+    if ($apiRequest['action'] === 'getcaselistheaders') {
+      if (
+        CRM_Core_Permission::check('basic case information') &&
+        !CRM_Core_Permission::check('administer CiviCase') &&
+        !CRM_Core_Permission::check('access my cases and activities') &&
+        !CRM_Core_Permission::check('access all cases and activities')
+      ) {
+        foreach ($result['values'] as $key => $header) {
+          if ($header['name'] === 'next_activity') {
+            unset($result['values'][$key]);
+          }
+        }
+      }
+    }
+
+    return $result;
+  }
+
+}
```

Comment:
🍰 Just a small comment for next time: When moving files from one place to another it's better to make every change in their own commit. In this case we should have fixed lint issues first and then moved the file. The reason for this is to make it easier to understand what changed.